### DISCUSSION TITLE
Storing instruction pointer history in prefetcher

### DISCUSSION THREAD

**Discussion by afschy** [ChampSim/ChampSim#482] [Q&A]
The prefetcher_cache_operate function only provides the instruction pointer of the instruction that triggered the access. I want to be able to access the instruction pointers of the last three instructions preceding the instruction that triggered the demand access. Is there a way to access and store the current instruction pointer from the prefetcher_cycle_operate function? Is there any other way to store instruction pointer history?

**Comment by ngober** [comment]
In general, no. The cache is not given information except for requests to it. Furthermore, in an out-of-order context, it's not quite clear what "three instructions preceding" a memory request might mean.
